File: src/main/java/eu/codetopic/utils/ui/activity/modular/module/CoordinatorLayoutModule.java
/*
 * Copyright 2017 Jiří Kuchyňka (Anty)
 *
 *    Licensed under the Apache License, Version 2.0 (the "License");
 *    you may not use this file except in compliance with the License.
 *    You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *    Unless required by applicable law or agreed to in writing, software
 *    distributed under the License is distributed on an "AS IS" BASIS,
 *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *    See the License for the specific language governing permissions and
 *    limitations under the License.
 */

package eu.codetopic.utils.ui.activity.modular.module;

import android.support.annotation.LayoutRes;
import android.view.View;
import android.view.ViewGroup;

import eu.codetopic.utils.R;
import eu.codetopic.utils.ui.activity.modular.ModularActivity;
import eu.codetopic.utils.ui.activity.modular.SimpleActivityCallBackModule;

/**
 * Created by anty on 10/13/17.
 *
 * @author anty
 */
public class CoordinatorLayoutModule extends SimpleActivityCallBackModule {

    @Override
    protected void onSetContentView(@LayoutRes final int layoutResID, SetContentViewCallBack callBack) {
        callBack.set(R.layout.cordinator_layout_base);
        callBack.addViewAttachedCallBack(() -> {
            ModularActivity activity = getActivity();
            activity.getLayoutInflater().inflate(layoutResID, activity.findViewById(R.id.base_coordinator_layout_content));
        });
    }

    @Override
    protected void onSetContentView(final View view, SetContentViewCallBack callBack) {
        callBack.set(R.layout.cordinator_layout_base);
        callBack.addViewAttachedCallBack(() -> {
            ((ViewGroup) getActivity().findViewById(R.id.base_coordinator_layout_content)).addView(view);
        });
    }

    @Override
    protected void onSetContentView(final View view, final ViewGroup.LayoutParams params, SetContentViewCallBack callBack) {
        callBack.set(R.layout.cordinator_layout_base);
        callBack.addViewAttachedCallBack(() -> {
            ((ViewGroup) getActivity().findViewById(R.id.base_coordinator_layout_content)).addView(view, params);
        });
    }
}

File: src/main/java/eu/codetopic/utils/ui/activity/modular/module/CoordinatorLayoutModule.kt
/*
 * utils
 * Copyright (C)   2017  anty
 *
 * This program is free  software: you can redistribute it and/or modify
 * it under the terms  of the GNU General Public License as published by
 * the Free Software  Foundation, either version 3 of the License, or
 * (at your option) any  later version.
 *
 * This program is distributed in the hope that it  will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied  warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
 * GNU General Public License for more details.
 *
 * You  should have received a copy of the GNU General Public License
 * along  with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package eu.codetopic.utils.ui.activity.modular.module

import android.support.annotation.LayoutRes
import android.view.View
import android.view.ViewGroup

import eu.codetopic.utils.R
import eu.codetopic.utils.ui.activity.modular.SimpleActivityCallBackModule

/**
 * Created by anty on 10/13/17.
 *
 * @author anty
 */
class CoordinatorLayoutModule : SimpleActivityCallBackModule() {

    override fun onSetContentView(@LayoutRes layoutResID: Int, callBack: SetContentViewCallBack) {
        callBack.set(R.layout.activity_module_coordinator_layout)
        callBack.addViewAttachedCallBack {
            activity.let {
                it.layoutInflater.inflate(layoutResID, it.findViewById(R.id.base_coordinator_layout_content))
            }
        }
    }

    override fun onSetContentView(view: View, callBack: SetContentViewCallBack) {
        callBack.set(R.layout.activity_module_coordinator_layout)
        callBack.addViewAttachedCallBack {
            activity.findViewById<ViewGroup>(R.id.base_coordinator_layout_content)
                    .addView(view)
        }
    }

    override fun onSetContentView(view: View, params: ViewGroup.LayoutParams,
                                  callBack: SetContentViewCallBack) {
        callBack.set(R.layout.activity_module_coordinator_layout)
        callBack.addViewAttachedCallBack {
            activity.findViewById<ViewGroup>(R.id.base_coordinator_layout_content)
                    .addView(view, params)
        }
    }
}
